### PR TITLE
Move DOM utility functions from @wordpress/blocks to @wordpress/dom

### PR DESCRIPTION
**Pull request by adamziel**
## Description
This PR moves a few DOM-specific functions from `@wordpress/blocks` package to `@wordpress/dom` package.

It's also worth noting that there are many transformers living in `@wordpress/blocks/api/raw-handling`, some specific to blocks and some very generic. The value of extracting generic ones is arguable as it would also split a single logic unit of code into a few separate ones. Let's agree it's out of scope for this PR and discuss separately if needed.

## How has this been tested?
Make sure all tests pass

## Types of changes
Non-breaking change

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

File: packages/blocks/src/api/raw-handling/index.js [hunk 1]
```diff
@@ -16,9 +16,16 @@ import figureContentReducer from './figure-content-reducer';
...
+
+/**
+ * WordPress dependencies
+ */
+import { getPhrasingContentSchema } from '@wordpress/dom';
+
+/**
+ * @deprecated since 5.6. Callers should import getPhrasingContentSchema directly from '@wordpress/dom' package.
+ */
 export { getPhrasingContentSchema };
```
Comment (youknowriad) [inline packages/blocks/src/api/raw-handling/index.js:29]: Was this a public API of "blocks". If it's the case, we can proxy it and add  a `deprecated` call?

**Review by youknowriad — COMMENTED** [review]
Seems like a good factorization. 👍 I'd love @ellatrix's opinion here if you have time.

**Review by ellatrix — APPROVED** [review]
Sounds good

File: packages/blocks/src/api/raw-handling/index.js (hunk 1 above)
Comment (adamziel) [reply packages/blocks/src/api/raw-handling/index.js:29]: I believe that's what I did, I will add updated README files to this PR

Comment (adamziel) [reply packages/blocks/src/api/raw-handling/index.js:29]: Ah, you meant the function call, not merely the annotation - sure!

**Pull request merged**